app/wasm/message_plugin.go

```diff
@@ -40,52 +44,70 @@ func (m *CustomMessenger) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddre
 		if err := json.Unmarshal(msg.Custom, &contractMsg); err != nil {
 			return nil, nil, sdkerrors.Wrap(err, "osmosis msg")
 		}
-		// if contractMsg.MintTokens != nil {
-		// 	return m.mintTokens(ctx, contractAddr, contractMsg.MintTokens)
-		// }
+		if contractMsg.MintTokens != nil {
+			return m.mintTokens(ctx, contractAddr, contractMsg.MintTokens)
+		}
 		if contractMsg.Swap != nil {
 			return m.swapTokens(ctx, contractAddr, contractMsg.Swap)
 		}
 	}
 	return m.wrapped.DispatchMsg(ctx, contractAddr, contractIBCPortID, msg)
 }
 
-// func (m *CustomMessenger) mintTokens(ctx sdk.Context, contractAddr sdk.AccAddress, mint *wasmbindings.MintTokens) ([]sdk.Event, [][]byte, error) {
-// 	err := PerformMint(m.bank, ctx, contractAddr, mint)
-// 	if err != nil {
-// 		return nil, nil, sdkerrors.Wrap(err, "perform mint")
-// 	}
-// 	return nil, nil, nil
-// }
-
-// func PerformMint(b *bankkeeper.BaseKeeper, ctx sdk.Context, contractAddr sdk.AccAddress, mint *wasmbindings.MintTokens) error {
-// 	if mint == nil {
-// 		return wasmvmtypes.InvalidRequest{Err: "mint token null mint"}
-// 	}
-// 	rcpt, err := parseAddress(mint.Recipient)
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	denom, err := GetFullDenom(contractAddr.String(), mint.SubDenom)
-// 	if err != nil {
-// 		return sdkerrors.Wrap(err, "mint token denom")
-// 	}
-// 	if mint.Amount.IsNegative() {
-// 		return wasmvmtypes.InvalidRequest{Err: "mint token negative amount"}
-// 	}
-// 	coins := []sdk.Coin{sdk.NewCoin(denom, mint.Amount)}
-
-// 	err = b.MintCoins(ctx, gammtypes.ModuleName, coins)
-// 	if err != nil {
-// 		return sdkerrors.Wrap(err, "minting coins from message")
-// 	}
-// 	err = b.SendCoinsFromModuleToAccount(ctx, gammtypes.ModuleName, rcpt, coins)
-// 	if err != nil {
-// 		return sdkerrors.Wrap(err, "sending newly minted coins from message")
-// 	}
-// 	return nil
-// }
+func (m *CustomMessenger) mintTokens(ctx sdk.Context, contractAddr sdk.AccAddress, mint *wasmbindings.MintTokens) ([]sdk.Event, [][]byte, error) {
+	err := PerformMint(m.tokenFactory, m.bank, ctx, contractAddr, mint)
+	if err != nil {
+		return nil, nil, sdkerrors.Wrap(err, "perform mint")
+	}
+	return nil, nil, nil
```

Comment:
Do we need to return `[]sdk.Event` and `[][]byte` if these are always nil?

Reply:
It's part of the required interface for DispatchMsg. And saves us from rewriting in the caller above.

That said, it would be interesting to define some standard data/events to return to contracts for better integration. But that would be a larger discussion with the Osmosis team on API design